tarofreighter/chain_porter.go

```diff
@@ -502,6 +549,111 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 	return nil
 }
 
+// timeSinceLastDeliveryAttempt calculates time duration which has elapsed since
+// the last delivery attempt.
+func timeSinceLastDeliveryAttempt(timestamps []time.Time) time.Duration {
+	// If there are no previous parcel delivery attempts, then we'll
+	// return early.
+	if len(timestamps) == 0 {
+		return time.Duration(0)
+	}
+
+	// Otherwise we'll select the latest timestamp and compute the surpassed
+	// time relative to the current time.
+
+	// Get the latest timestamp without assuming order.
+	latestTimestamp := timestamps[0]
```

Comment:
I guess that in the future we will use `QueryParcelDeliveryLog` to do more than just getting the latest timestamp (like print all the times that we tried to to send the proof in the cmd line or something) If not we can have directly a method that says "QueryLastParcelDeliveryLog" that directly returns the latest timestamp.

Reply:
Yes, I figured that in the future some UI might need to know all timestamps. So I went with the more general query.